src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/AldersvilkårVurdering.kt

```diff
@@ -0,0 +1,83 @@
+package no.nav.tilleggsstonader.sak.vilkår.vilkårperiode
+
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.Feil
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.feilHvis
+import no.nav.tilleggsstonader.sak.opplysninger.grunnlag.Grunnlagsdata
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.MålgruppeType
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.faktavurderinger.SvarJaNei
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.dto.LagreVilkårperiode
+import java.time.LocalDate
+
+object AldersvilkårVurdering {
+    fun vurderAldersvilkår(
+        vilkårperiode: LagreVilkårperiode,
+        grunnlagsData: Grunnlagsdata,
+    ): SvarJaNei {
+        val fødselsdato = grunnlagsData.grunnlag.fødsel?.fødselsdato
+
+        feilHvis(fødselsdato == null) { "Kan ikke vurdere aldersvilkår uten å vite fødselsdato til bruker" }
+
+        val gyldig: SvarJaNei =
+            when (vilkårperiode.type as MålgruppeType) {
+                MålgruppeType.AAP, MålgruppeType.NEDSATT_ARBEIDSEVNE, MålgruppeType.UFØRETRYGD ->
+                    vurderAldersvilkårForNedsattArbeidsevne(fødselsdato, vilkårperiode)
+                MålgruppeType.OMSTILLINGSSTØNAD -> vurderAldersvilkårForOmstillingsstønad(fødselsdato, vilkårperiode)
+                MålgruppeType.OVERGANGSSTØNAD -> SvarJaNei.JA_IMPLISITT
+                else -> throw Feil("Aldersvilkår vurderes ikke for målgruppe: ${vilkårperiode.type}")
+            }
+        return gyldig
```

Comment:
Kanskje mest preferanse men her kan man retunere when direkte. Også liker jeg best at ikke typer opp variabler eller caster typer uten at man må

```suggestion
 return when (vilkårperiode.type) {
            MålgruppeType.AAP, MålgruppeType.NEDSATT_ARBEIDSEVNE, MålgruppeType.UFØRETRYGD ->
                vurderAldersvilkårForNedsattArbeidsevne(fødselsdato, vilkårperiode)
            MålgruppeType.OMSTILLINGSSTØNAD -> vurderAldersvilkårForOmstillingsstønad(fødselsdato, vilkårperiode)
            MålgruppeType.OVERGANGSSTØNAD -> SvarJaNei.JA_IMPLISITT
            else -> throw Feil("Aldersvilkår vurderes ikke for målgruppe: ${vilkårperiode.type}")
        }
        return gyldig
```

Reply:
Ser den! Endret her: a06e480982af580c75fd4d268b64070332a9e6c8